src/Algebra/Morphism/TwoMagmas.agda

```diff
@@ -0,0 +1,56 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some properties of Magma homomorphisms
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+
+open import Algebra using (Magma)
+import Algebra.Morphism.Definitions as Definitions
+import Algebra.Morphism.ToSetoid as ToSetoid
+open import Function using (id; _∘_)
+open import Relation.Binary using (_Preserves_⟶_)
+import Relation.Binary.EqReasoning as EqR
+open import Relation.Unary using (Pred)
+
+
+module Algebra.Morphism.TwoMagmas {α α= β β=} (M : Magma α α=) (M' : Magma β β=)
```

Comment:
After ditching the `Magma` module parameters (described how below) I'd then rename this module `Algebra.Morphism.Consequences` which is more standardised.

Reply:
This concerns several your last comments.
Please, consider once more the above subject of                                   
Morphism.ToSetoid and Morphism.TwoMagmas.                                         
                                                                                  
* It is strange the you are against `_≈∘_`.                                      
  It is a very natural generalization for `_≗_`,   a generally needed notion.                              
  Its context requires a domain of Set and a codomain of Setoid -- only this.     
  It is evident that one day Standard will introduce such a relation on functions.                                                                      
  With this, there also naturally appears the corresponding Setoid instance on    
  `A → B`.  Generally, a useful thing. You are going to remove this instance.       
  Well, at least `_≈∘_` is used in TwoMagmas.                                       
                                                                                  
* To formulate and to prove the lemma `IsHomo⇒IsHomo-rev`,                      
  there are needed                                                                
  a) two Setoids, b) binary operations ``_∙_, _∙'_`` on these setoids,                
  c) the congruence property for `_∙'_`.                          
                                                                                  
All these requirements together mean just _two magmas_                            
(with adding only a single condition of congruence for  `_∙_`).                       
This is why it is introduced as                                                   
```                                                                               
 module Algebra.Morphism.TwoMagmas ... (M : Magma α α=) (M' : Magma β β=)                       
```                                                                                                
As to f and g in the signature, it is required congruence for g.                  
                                                                                  
Please, see  Algebra.Morphism.TwoMagmas.                                          
It is changed to meet somewhat half of your requirements for this subject.
                                                                                  
You suggest to use `IsInverse` in Function.Structures.                            
It consists of  `isLeftInverse` and  `inverseʳ`. 
And this requires the congruence property for  f.                                  
And this property is not required in the lemma.                                   
So, I use instead AreInverseMaps defined in Algebra.Morphism.TwoMagmas.           
                                                                                  
Please, advice.

Reply:
> It is a very natural generalization for _≗_, a generally needed notion.

Yes, you're right, it is useful. However a) `Algebra.Morphism.ToSetoid` is not the right place to put it and b) it's not needed by this PR and c) this PR is already huge enough as it is. The smaller this PR is the greater the chance of it getting merged in.

> You suggest to use IsInverse in Function.Structures.
It consists of isLeftInverse and inverseʳ.
And this requires the congruence property for f.
And this property is not required in the lemma.

Hmm well I assumed that adding an extra congruency assumption isn't too harmful. For example you've already done so yourself, note that you've assumed congruency of `_∙_` via `Magma` but not used it. I'm happy with your current solution of using `Inverseᵇ` however.